packages/instantsearch.js/src/widgets/index/__tests__/index-test.ts

```diff
@@ -2894,6 +2902,64 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/index-widge
     });
   });
 
+  describe('setIndexUiState', () => {
+    it('updates main ui state with object', () => {
+      const instance = index({ indexName: 'indexName' });
+      const instantSearchInstance = instantsearch({
+        indexName: 'root',
+        searchClient: createSearchClient(),
+      });
+      instantSearchInstance.start();
+      instantSearchInstance.addWidgets([
+        instance.addWidgets([virtualSearchBox({})]),
+      ]);
+
+      instance.setIndexUiState({
+        query: 'iphone',
+      });
+
+      expect(instantSearchInstance.getUiState()).toEqual({
+        root: {},
+        indexName: {
+          query: 'iphone',
+        },
+      });
+    });
+
+    it('updates main ui state with function', () => {
+      const instance = index({ indexName: 'indexName' });
+      const instantSearchInstance = instantsearch({
+        indexName: 'root',
+        searchClient: createSearchClient(),
+      });
+      instantSearchInstance.start();
+      instantSearchInstance.addWidgets([
+        instance.addWidgets([virtualPagination({})]),
+      ]);
+      instance.setIndexUiState((uiState) => ({
+        page: (uiState.page || 1) + 1,
+      }));
+
+      expect(instantSearchInstance.getUiState()).toEqual({
+        root: {},
+        indexName: {
+          page: 2,
+        },
+      });
+
+      instance.setIndexUiState((uiState) => ({
+        page: (uiState.page || 1) + 1,
+      }));
```

Comment:
Is the user responsible for merging with the existing UI state if they want to preserve other keys, or do we merge?

Reply:
just like the main setUiState, the user is merging